go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -214,6 +300,63 @@ func createPerNodePhysicalVIPs(isIPv6 bool, protocol v1.Protocol, sourcePort int
 	return nil
 }
 
+// createPerNodePhysicalVIPs adds load balancers on a per node basis for GR and worker switch LBs using physical IPs
+func createPerNodePhysicalVIPsLocal(isIPv6 bool, protocol v1.Protocol, sourcePort int32, slices []*discovery.EndpointSlice, svcPort v1.ServicePort, family v1.IPFamily) error {
```

Comment:
make this function just take in the previously referenced map[nodeName]LbEndpoints, then you can just select them based on the GR you are iterating on in this function and reduce the number of args you have to pass here.